### ISSUE TITLE
Fix discussion list slide gestures on non-mobile touch screens (e.g. iPad)

### ISSUE THREAD

**Issue by tobscure** [flarum/core#896]

**Comment by jordanjay29** [comment]
What is this? Description please?

**Comment by luceos** [comment]
I assume, reading the title again, that touch control is only enabled on mobile devices (with smaller screen sizes) whereas these should also be enabled on larger touch-enabled devices like iPads.

**Comment by tobscure** [comment]
Sorry, other way around - touch gestures are enabled on all touch screens, but they are only styled properly on smaller screen sizes. So if you swipe a discussion on iPad, funky stuff happens :O

**Comment by datitisev** [comment]
Demo:

![2ab0b2ea-c967-448b-8892-aaacdd154114](https://user-images.githubusercontent.com/6401250/39919529-e6819456-54e1-11e8-8223-04262c7b7df9.jpeg)

**Comment by TurboProgramming** [comment]
[This issue needs the `mobile` tag]

**Comment by ardacebi** [comment]
I have a solution to this, but first I need to know something.

Is there a way we can make Flarum detect if the client is a mobile device (touchscreen) or a computer (non-touchscreen)?

PS: Here, I don't mean detection by screen size, like can we recognize the screen itself.

**Comment by datitisev** [comment]
@ardacebi CSS or JS?

With CSS one can use the `screen` media like
```css
@media screen {}
```

**Comment by ardacebi** [comment]
Doesn't matter, as I understand: you can.

Then I was going to suggest: if the user is using a mobile device like the iPad, let's use the vertical layout of Flarum where tags and other buttons move to the top of the discussion list, scrollable, giving space to the discussion list for them to slide without interfering with other elements.

![screenshot_2018-08-13 flarum community 1](https://user-images.githubusercontent.com/17576065/44006968-bce9b7b0-9e95-11e8-8ee3-fd1a2e127269.png)

**Comment by tobscure** [comment]
@ardacebi we already do adapt the layout according to the screen size

**Comment by ardacebi** [comment]
@tobscure No, what I mean is different.

Flarum already recognizes touch screens and adds that gesture thing. But the thing is when using iPad (or related devices) in landscape mode, it displays like on desktop, tags, and buttons on the left side, discussions on the right side. That causes that funky stuff to happen.

I'm saying that let's use the vertical layout I sent you above on "all" touch screens, not just vertical iPad but also on horizontal iPad.

Are you getting what I'm saying?

**Comment by tobscure** [comment]
I get what you're saying and I disagree. The horizontal iPad screen is wide enough to display the sidebar so it should. The funky behaviour with the gestures is a bug.

**Comment by ardacebi** [comment]
I may have another fix while displaying the sidebar. I’ll open a PR.

**Comment by luceos** [comment]
@ardacebi we would be appreciative would you first bounce the idea here 👍

**Comment by ardacebi** [comment]
@luceos Instead of just setting the secondary color to the first area where the extension was located, I will apply the secondary color to itself. So all elements on it will be visible. I may also add a box shadow if it looks good 🤔

But not sure about the shadow.

**Comment by luceos** [comment]
If I'm not mistaken the original issue is about non-mobile -yet- handheld devices using part of the mobile functionality, but not it's styling. From what I can tell from the screenshot the discussion entry shifts away from the list, changing changing layout or looks therefore is not a solution and should not be discussed in this issue.

Please do correct me if I'm wrong.

**Comment by ardacebi** [comment]
You're wrong and right. The issue is about tablet styled devices (not phones). Giving them a background will solve the whole issue as it's overlapping the sidebar without a background giving it a bad looking.

If you look at @datitisev's screenshot, you will see that the background layer stays where it was not following the discussion item itself. But if the discussion has a background (Secondary color in this case), it won't overlap the sidebar text and maybe with a box shadow, we may create a better look (idea from search box issue).
